src/test/components/TooltipMarker.test.js

```diff
@@ -361,6 +363,23 @@ describe('TooltipMarker', function() {
       [
         'FileIO (non-main thread)',
         114.5,
+        {
+          type: 'FileIO',
+          startTime: 114,
+          endTime: 115,
+          source: 'PoisonIOInterposer',
+          filename: '/foo/bar',
+          operation: 'create/open',
+          cause: {
+            time: 17.0,
+            stack: funcNames.indexOf('nsRefreshDriver::AddStyleFlushObserver'),
+          },
+          threadId: threadId,
```

Comment:
Isn't it a bit incorrect that you're using the threadId for the main thread even though you say that it's for a non main thread? I think you should add another thread here. Maybe it would be easier to split this specific test to a separate test block. You can use `getProfileWithMarkers` from `src/test/fixtures/profiles/processed-profile.js` to generate such a profile easily (accepts several arguments to generate several threads), then change the generated threads to add the threadId/threadName values you want

Reply:
I thought it would be easier to add it here at first because it doesn't matter if the given `threadId` is the same with current thread, implementation wise. But you are right about it being not completely accurate with real life example. Split it to a new test block and added another thread to reference to.